### PR TITLE
ZCS-15041 : Adjust the installer script to accommodate the updated licensing requirements

### PR DESCRIPTION
**Pull request by umagmrit**
- Earlier, we were using `zmlicense --ping`, `curl`, and `wget` to check connectivity to the license server, but with the new license changes, the `--ping` option was removed from the `zmlicense`, so we are going to use `curl` and `wget` only to verify connections to the license server.

- Checking for an available license using the `zmlicense -p` if the `zmlicense` utility is available. If `zmlicense` is not available or the license is not installed, the installer will get the existing license details from the `/opt/zimbra/conf/ZCSLicense.xml` or `/opt/zimbra/conf/ZCSLicense-Trial.xml` files. If these files do not exist, the installer will abort the upgrade with an error, but with a new change, the installer will allow the upgrade by giving a warning `The ZCS Network upgrade requires a valid license to be installed`.
- Delete `/opt/zimbra/conf/ZCSLicense.xml` and `/opt/zimbra/conf/ZCSLicense-Trial.xml` after installation/upgrade.

### CODE REVIEW

File: rpmconf/Install/Util/utilfunc.sh [hunk 1]
```diff
@@ -979,56 +980,42 @@ verifyLicenseActivationServer() {
...
-    elif [ ${ZM_CUR_MAJOR} -gt "7" ]; then
-      /opt/zimbra/bin/zmlicense --ping > /dev/null 2>&1
-    else
-      /opt/zimbra/java/bin/java -XX:ErrorFile=/opt/zimbra/log -client -Xmx256m -Dzimbra.home=/opt/zimbra -Djava.library.path=/opt/zimbra/lib  -classpath ./lib/jars/zimbra-license-tools.jar:/opt/zimbra/lib/jars/* com.zimbra.cs.license.LicenseCLI --ping > /dev/null 2>&1
-    fi
-    if [ $? != 0 ]; then
-      activationWarning
-    fi
+  # make sure we can contact the activation server for automated activation
+  echo $HOSTNAME | egrep -qe 'vmware.com$|zimbra.com$|zimbradev.com$' > /dev/null 2>&1
```
Comment (prashantajari) [inline rpmconf/Install/Util/utilfunc.sh]: We should remove vmware.com references

**Review by RakeshAMore — APPROVED** [review]

**Review by shrutig0510 — APPROVED** [review]

File: rpmconf/Install/Util/utilfunc.sh [hunk 2]
```diff
@@ -1070,14 +1054,20 @@ verifyLicenseAvailable() {
...
+	  else
+		  licensedUsers=`su - zimbra -c "zmlicense -p | grep ^AccountsLimit | sed -e 's/AccountsLimit=//'" 2> /dev/null`
+		  licenseValidUntil=`su - zimbra -c "zmlicense -p | grep ^ValidUntil= | sed -e 's/ValidUntil=//'" 2> /dev/null`
+		  licenseType=`su - zimbra -c "zmlicense -p | grep ^InstallType= | sed -e 's/InstallType=//'" 2> /dev/null`
+	  fi
   fi
 
   # parse files if license tool wasn't there or didn't return a valid license
-  if [ x"$licenseCheck" = "xlicense not installed" -o x"$licenseCheck" = "x" ]; then
+  if [ x"$licenseCheck" = "xlicense not installed" ] || [ x"$licenseCheck" = "x" ] || [[ "$licenseCheck" =~ "License is not activated" ]]; then
```
Comment (prashantajari) [inline rpmconf/Install/Util/utilfunc.sh:1070]: Why we have added "License is not activated" condition?

Comment (umagmrit) [reply rpmconf/Install/Util/utilfunc.sh:1070]: "License is not activated", which is the output of new zmlicense changes. @RakeshAMore correct me if I am wrong.

**Review by prashantajari — APPROVED** [review]
Looks good.

File: rpmconf/Install/Util/utilfunc.sh (hunk 2 above)
Comment (RakeshAMore) [reply rpmconf/Install/Util/utilfunc.sh:1070]: Yes this is correct. If license is not activated.
https://github.com/Zimbra/zm-license-tools/blob/develop/src/java/com/zimbra/cs/license/LicenseCLI.java#L167

**Pull request merged**